db/system_keyspace_clients.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "system_keyspace_clients.hh"
+#include "system_keyspace.hh"
+#include "schema_builder.hh"
+#include "query_context.hh"
+#include "query-result-set.hh"
+#include "service/client_state.hh"
+#include <stdexcept>
+
+namespace db::system_keyspace {
+
+extern logging::logger slogger;
+
+sstring to_string(client_type ptype) {
+    switch (ptype) {
+        case client_type::cql: return "cql";
+        case client_type::thrift: return "thrift";
+        case client_type::alternator: return "alternator";
+        default: throw std::runtime_error("Invalid client_type");
+    }
+}
+
+/** Layout based on C*-4.0.0 with extra columns `shard_id' and `client_type'
+  * but without `request_count'. Also CK is different: C* has only (`port'). */
+schema_ptr clients() {
+    thread_local auto clients = [] {
+        schema_builder builder(make_lw_shared(schema(generate_legacy_id(NAME, CLIENTS), NAME, CLIENTS,
+            // partition key
+            {{"address", inet_addr_type}},
+            // clustering key
+            {{"client_type", utf8_type}, {"port", int32_type}},
```

Comment:
I think we should swap these two, so a `SELECT * FROM system.clients WHERE address = '1.2.3.4' AND port = 1234` works without ALLOW FILTERING. Usually there will be only one client type, so selecting by client type is less interesting.